src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -127,9 +129,10 @@ protected override ModuleData CreateValueFromKey(string key)
         }
         private SimpleNameHashtable _simpleNameHashtable = new SimpleNameHashtable();
 
-        public CompilerTypeSystemContext(TargetDetails details)
+        public CompilerTypeSystemContext(TargetDetails details, bool multifileCompilation)
```

Comment:
Mmmm, maybe instead of Compilation.cs, this should be on NodeFactory. After all, this logic mostly drives the graph generation.